vcd/resource_vcd_vapp_vm.go

```diff
@@ -123,6 +137,34 @@ func vmSchemaFunc(vmType typeOfVm) map[string]*schema.Schema {
 			Description:  "The amount of RAM (in MB) to allocate to the VM",
 			ValidateFunc: validateMultipleOf4(),
 		},
+		"memory_reservation": &schema.Schema{
+			Type:         schema.TypeInt,
+			Optional:     true,
+			Computed:     true,
+			Description:  "The amount of RAM (in MB) reservation on the underlying virtualization infrastructure",
+			ValidateFunc: validateMultipleOf4(),
+		},
+		"memory_priority_type": &schema.Schema{
+			Type:         schema.TypeString,
+			Optional:     true,
+			Computed:     true,
+			Description:  "Pre-determined relative priorities according to which the non-reserved portion of this resource is made available to the virtualized workload",
+			ValidateFunc: validation.StringInSlice([]string{"LOW", "NORMAL", "HIGH", "CUSTOM"}, false),
+		},
+		"memory_shares": &schema.Schema{
+			Type:         schema.TypeInt,
+			Optional:     true,
+			Computed:     true,
+			Description:  "Custom priority for the resource. This is a read-only, unless the share level is CUSTOM",
+			ValidateFunc: validateMultipleOf4(),
+		},
+		"memory_limit": &schema.Schema{
+			Type:         schema.TypeInt,
+			Optional:     true,
+			Computed:     true,
+			Description:  "The limit for how much of memory can be consumed on the underlying virtualization infrastructure. This is only valid when the resource allocation is not unlimited.",
+			ValidateFunc: validateMultipleOf4(),
```

Comment:
I see that there is a "unlimited" checkbox in UI and when it is ticked the request sends `-1`. In our case we have this validation and it fails immediatelly:
```
╷
│ Error: expected memory_limit to be multiple of 4, got -1
│ 
│   with vcd_vm.TestAccVcdVmAdvancedComputeProperties-18400,
│   on main.tf line 41, in resource "vcd_vm" "TestAccVcdVmAdvancedComputeProperties-18400":
│   41:   memory_limit         = -1
│ 
╵
```


<img width="539" alt="image" src="https://user-images.githubusercontent.com/15804230/151536268-c5b62609-a60f-40ed-8019-1da743061dc7.png">